File: examples/ruby-rspec/lib/roman_numerals.rb
# frozen_string_literal: true

module RomanNumerals
  NUMERALS = [
    ["M", 1000], ["CM", 900], ["D", 500], ["CD", 400], ["C", 100], ["XC", 90],
    ["L", 50], ["XL", 40], ["X", 10], ["IX", 9], ["V", 5], ["IV", 4], ["I", 1]
  ].freeze

  module_function

  def to_roman(num)
    result = +""
    NUMERALS.each do |letter, value|
      while num >= value
        num -= value
        result << letter
      end
    end
    result
  end
end

File: examples/kotlin-kotest/src/main/kotlin/examples/RomanNumerals.kt
package examples

private val NUMERALS =
    listOf(
        "M" to 1000,
        "CM" to 900,
        "D" to 500,
        "CD" to 400,
        "C" to 100,
        "XC" to 90,
        "L" to 50,
        "XL" to 40,
        "X" to 10,
        "IX" to 9,
        "V" to 5,
        "IV" to 4,
        "I" to 1,
    )

fun toRoman(num: Int): String {
    var remaining = num
    val result = StringBuilder()
    for ((letter, value) in NUMERALS) {
        while (remaining >= value) {
            remaining -= value
            result.append(letter)
        }
    }
    return result.toString()
}
